### PR TITLE
Selector structure refactoring

### PR DESCRIPTION
**Pull request by chadwpry**
Implement a new model for structuring selectors. This will allow for the selector, but also include an attribute to read from the selected node.

### CODE REVIEW

File: extension/content.js [hunk 1]
```diff
@@ -1,11 +1,12 @@
-let constructPosition = require('./lib/position.js').default;
+let $ = require('jquery');
```
Comment (robskrob) [inline extension/content.js]: I don't see the `$` being used anywhere in this file. Are you sure we need it?

File: extension/content.js [hunk 2]
```diff
@@ -1,11 +1,12 @@
...
+let position = require('./lib/position.js').default;
 
 chrome.runtime.onMessage.addListener(
   function(request, sender, sendResponse) {
     console.log('content.js', 'receive message');
 
     if (request.type === "capture_position") {
-      constructPosition(function(position) {
+      position.construct(function(position) {
```
Comment (robskrob) [inline extension/content.js]: we may want to change the name of the callbacks argument to, `object`

Comment (robskrob) [inline extension/content.js]: Eventually, we are probably going to rename the file `lib/position`, the function, `position` and its callback's argument, `position`.

File: extension/lib/position.js [hunk 3]
```diff
@@ -1,52 +1,50 @@
...
+let construct = function(callback) {
   let url      = 'https://localhost/api/v1/selectors';
   let promise  = $.getJSON(url, {
     host: location.host
   });
 
   promise.done(function(json, status, xhr) {
-    let position = Object.keys(json.data).reduce(function(position, key) {
-      let elements = $(json.data[key]);
+    let object = Object.keys(json.data).reduce(function(object, key) {
```
Comment (robskrob) [inline extension/lib/position.js]: For the selector resource, aren't the keys the css selectors? It looks like the client wants keys in `json.data[key]` to return an object with a key, `selector`, which will contain the css selector, and another key, `attribute`, which will return the value of an elements HTML attribute.

File: routes/api/selectors.js [hunk 4]
```diff
@@ -7,16 +7,50 @@ router.get('/', function(req, res, next) {
```
Comment (robskrob) [inline routes/api/selectors.js:7]: Looks good. I'll just have to reseed the data on my side in gcloud.

**Comment by robskrob** [comment]
@chadwpry left some comments on the PR.

File: extension/content.js (hunk 1 above)
Comment (chadwpry) [reply extension/content.js]: Really good point. It is only used in the position file now.

File: extension/content.js (hunk 2 above)
Comment (chadwpry) [reply extension/content.js]: Yep > I am now changing it to lib/capture.js. I a better structure will come together as it is built out. The position is now changed to product.

File: extension/lib/position.js (hunk 3 above)
Comment (chadwpry) [reply extension/lib/position.js]: The key is what we are attempting to structure, the selector is the css selector to the dom node, and the attribute is the attribute on the dom node that returns the information.

I changed it to be more flexible. Here is an example...

IMG -> img.src
META -> meta.content
H1 -> h1.innerText
INPUT -> input.value

They are all different attributes, but the existing selector structure would have required the code to include a lot of if statements to handle "types" of nodes in different ways. In the way this is written, we can now offload that condition logic into the structure of the selector and make the code use that logic directly.

File: routes/api/selectors.js (hunk 4 above)
Comment (chadwpry) [reply routes/api/selectors.js:7]: Ok. When I used sequelize, it allows for rails like db/seed and db/migration. Does that exist in gcloud or do you have to build that manually?

**Comment by robskrob** [comment]
👍

**Pull request merged**